src/test/java/io/jenkins/plugins/analysis/warnings/IssuesRecorderITest.java

```diff
@@ -176,4 +211,113 @@ private AnalysisResult scheduleBuildAndAssertStatus(final FreeStyleProject job,
+            for (ResultAction elements : actions) {
+                results.add(elements.getResult());
+            }
+
+            return results;
+        }
+        catch (Exception e) {
+            throw new AssertionError(e);
+        }
+    }
+
+    /**
+     * Runs the CheckStyle and PMD tools on an output file that contains several issues: the build should report 6 and 4
+     * issues.
+     */
+    @Test
+    @WithTimeout(1000)
+    public void shouldCreateMultipleToolsAndAggregationResultWithWarningsAggregateFalse() {
+        FreeStyleProject project = createJobWithWorkspaceFile("checkstyle.xml", "pmd-warnings.xml");
+        enableWarningsAggregation(project, false, "**/checkstyle-issues.txt", new CheckStyle(),
+                "**/pmd-warnings-issues.txt", new Pmd());
+
+        List<AnalysisResult> results = scheduleBuildAndAssertStatusForBothTools(project, Result.SUCCESS);
+
+        for (AnalysisResult element : results) {
+            if (element.getId().equals("checkstyle")) {
+                assertThat(element.getTotalSize()).isEqualTo(6);
+            }
+            else {
+                assertThat(element.getId()).isEqualTo("pmd");
+                assertThat(element.getTotalSize()).isEqualTo(4);
+            }
+            assertThat(element).hasOverallResult(Result.SUCCESS);
+        }
+    }
+
+    /**
+     * Runs the CheckStyle and PMD tools on an output file that contains one issue: the build should report 10 issues.
+     */
+    @Test
+    @WithTimeout(1000)
+    public void shouldCreateMultipleToolsAndAggregationResultWithWarningsAggregateTrue() {
+        FreeStyleProject project = createJobWithWorkspaceFile("checkstyle.xml", "pmd-warnings.xml");
+        enableWarningsAggregation(project, true, "**/checkstyle-issues.txt", new CheckStyle(),
+                "**/pmd-warnings-issues.txt", new Pmd());
+
+        AnalysisResult result = scheduleBuildAndAssertStatus(project, Result.SUCCESS);
+
+        assertThat(result).hasTotalSize(10);
+        assertThat(result.getSizePerOrigin()).containsKeys("checkstyle", "pmd");
+        assertThat(result).hasOverallResult(Result.SUCCESS);
+    }
+
+    /**
+     * Runs only the CheckStyle tool multiple times on an output file that contains not several issues and produce a
+     * failure.
+     */
+    @Test
+    @WithTimeout(1000)
+    public void shouldCreateMultipleToolsAndAggregationResultWithWarningsAggregateFalseAndSameTool() {
+        FreeStyleProject project = createJobWithWorkspaceFile("checkstyle2.xml", "checkstyle3.xml");
+        enableWarningsAggregation(project, false, "**/checkstyle2-issues.txt", new CheckStyle(),
+                "**/checkstyle3-issues.txt", new CheckStyle());
+
+        List<AnalysisResult> results = scheduleBuildAndAssertStatusForBothTools(project, Result.FAILURE);
+
+        for (AnalysisResult elements : results) {
+            assertThat(elements).hasErrorMessages();
+        }
+    }
```

Comment:
Hier fehlt noch der Check, dass eine der Actions nicht hinzugefügt werden kann, d.h. im Console Log steht der Text: 
```
java.lang.IllegalStateException: ID checkstyle is already used by another action: io.jenkins.plugins.analysis.core.views.ResultAction for CheckStyle
```

Reply:
Erledigt. Prüfung ob Size gleich 1, zusätzlich ob vom Typ "checkstyle".

Reply:
Zusatz: Diese Warning wird bei mir nicht in der Konsole ausgegeben.

Reply:
Im Console Log des Builds der Jenkins Instanz! 

```
j.assertLogContains("ID checkstyle is already used by another action: io.jenkins.plugins.analysis.core.views.ResultAction for CheckStyle", build)
```

Reply:
Ok. Habe ich so eingebaut.